simulation/container.py

```diff
@@ -0,0 +1,181 @@
+
+        Example
+        -------
+        >>> container.count_dead()
+        """
+
+        return super().count_objects_in_list(groups.DGroup)
+
+    def initial_set_up(self, susceptible: int, infected: int, recovered: int,
+                       dead: int, ) -> None:
+        """
+        Method which specified amount of specific population group in object_list.
+        When sum of groups is greater than population parameter raise ValueError.
+
+        Parameters
+        ----------
+        susceptible: int, required
+            Amount of susceptible instances to place in container.
+        infected: int, required
+            Amount of infected instances to place in container.
+        recovered: int, required
+            Amount of infected instances to place in container.
+        dead: int, required
+            Amount of dead instances to place in container.
+
+        Example
+        -------
+        >>> container.initial_set_up(90, 10, 0, 0)
+        """
+
+        if susceptible + infected + recovered + dead > self.population:
+            raise ValueError("Sum of population groups is greater than population")
+
+        self.add_instances(self.smember, susceptible)
+        self.add_instances(self.imember, infected)
+        self.add_instances(self.rmember, recovered)
+        self.add_instances(self.dmember, dead)
+
+    def simulation(self) -> None:
+        """
+        Core logic of container instance contains path of live from susceptible
+        instance to recover or dead instance.
+        One cycle is:
+            - instances moves
+            - if susceptible instance is in infected instance infection area
+              check if infect
+            - if is infected remove this instance from object list and add
+              new infected instance
+            - check if infected instance get recovered, if true removed this
+              instance and add new recovered instance
+            - check if infected instance died, if true removed this instance
+              and add new dead instance
+        """
+
+        while self.is_alive():
+            for instance in self.object_list:
+                x = random.uniform(-self.move_distance_length, self.move_distance_length)
+                y = random.uniform(-self.move_distance_length, self.move_distance_length)
+                instance.move(x, y, self.width, self.height)
+
+                if isinstance(instance, groups.SGroup):
+                    for infected in self.object_list:
+                        if isinstance(infected, groups.IGroup):
+                            if instance.is_in_infection_area(infected.x, infected.y,
+                                                             infected.infection_range):
+                                if instance.infect():
+                                    self.object_list.remove(instance)
+                                    self.add_instances(self.imember, 1)
```

Comment:
The question is, do we preserve individual names/ids that way?
Eg. during visualization we would want to show particular individuals, after such removal and new object creation there will be weird consequences probably. We should know after some visualization implementation.